crates/autopilot/src/infra/persistence/mod.rs

```diff
@@ -63,27 +63,52 @@ impl Persistence {
         LeaderLock::new(self.postgres.pool.clone(), key, Duration::from_millis(200))
     }
 
-    /// There is always only one `current` auction.
-    ///
-    /// This method replaces the current auction with the given one.
-    ///
-    /// If the given auction is successfully saved, it is also archived.
-    pub async fn replace_current_auction(
-        &self,
-        auction: &domain::RawAuctionData,
-    ) -> Result<domain::auction::Id, DatabaseError> {
-        let _timer = observe::metrics::metrics()
-            .on_auction_overhead_start("autopilot", "replace_auction_in_db");
-        let auction = dto::auction::from_domain(auction.clone());
+    /// Fetches the ID that should be used for the next auction.
+    pub async fn get_next_auction_id(&self) -> Result<domain::auction::Id, DatabaseError> {
+        let _timer = Metrics::get()
+            .database_queries
+            .with_label_values(&["get_next_auction_id"])
+            .start_timer();
         self.postgres
-            .replace_current_auction(&auction)
+            .get_next_auction_id()
             .await
-            .inspect(|&id| {
-                self.archive_auction(dto::auction::Auction { id, auction });
-            })
             .map_err(DatabaseError)
     }
 
+    /// Spawns a background task that replaces the current auction in the DB
+    /// with the new one.
+    pub fn replace_current_auction_in_db(
+        &self,
+        id: domain::auction::Id,
+        auction: &domain::RawAuctionData,
```

Comment:
I think the variables could have better names, it's kind of hard to tell which is which by just looking at it. Is the ID the new one, or is it contained in AuctionData?